### PR TITLE
[Rel Eng] Dial In Test Skipping

### PR DESCRIPTION
**Pull request by robertgshaw2-neuralmagic**
SUMMARY:
* updated model test structure to focus on core models
* refactored tests to use environment variables (currently at "test group" level - so each folder has an env variable). All tests are off by default and they are explicitly enabled
* refactored workflows `build-test` workflow to use a list of env variables rather than skip test list

WHY:
* this enables us to be more sane about what is and is not on - as opposed to a long list of files
* this enables us to actually track what is run and what is not run (via testmo, which tracks skipped tests)
* this enables us to have more fine-grained control over what is run vs not run (we can add more env vars at the sub-group level to turn off more tests)

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def __init__(
         self,
         model_name: str,
         dtype: str = "half",
-        access_token: Optional[str] = None,
+        **kwargs,
```
Comment (robertgshaw2-neuralmagic) [inline tests/conftest.py:150]: - access token not needed (we set `HF_TOKEN` env variable in automation)
- kwargs lets us pipe anything through to HF constructor

File: tests/nm_utils/server.py [hunk 2]
```diff
@@ -10,8 +9,6 @@
 import requests
 import torch
 
-from tests.nm_utils.logging import log_banner
```
Comment (robertgshaw2-neuralmagic) [inline tests/nm_utils/server.py:13]: Note: bad import?

File: neuralmagic/tests/test_skip_env_vars/nightly.txt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+TEST_ACCURACY=0
```
Comment (robertgshaw2-neuralmagic) [inline neuralmagic/tests/test_skip_env_vars/nightly.txt:1]: update these to use "enable" or "disable"

File: tests/nm_utils/utils_skip.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    "TEST_SAMPLERS": should_skip_samplers_test_group,
+    "TEST_SPEC_DECODE": should_skip_spec_decode_test_group,
+    "TEST_TENSORIZER_LOADER": should_skip_tensorizer_loader_test_group,
+    "TEST_TOKENIZATION": should_skip_tokenization_test_group,
+    "TEST_WORKER": should_skip_worker_test_group,
+}
+
+
+def should_skip_test_group(group_name: str) -> bool:
+    return MAP[group_name]()
```
Comment (dhuangnm) [inline tests/nm_utils/utils_skip.py:134]: Why do we need to return the map here? Should we just check the env var based on the group_name and return true/false? This way we don't need the group_name specific should_skip_xxx() functions above?

File: tests/async_engine/test_api_server.py [hunk 5]
```diff
@@ -7,6 +7,12 @@
 import pytest
 import requests
 
+from tests.nm_utils.utils_skip import should_skip_test_group
+
+if should_skip_test_group(group_name="TEST_ASYNC_ENGINE"):
+    pytest.skip("TEST_ASYNC_ENGINE=0, skipping async engine test group",
```
Comment (dhuangnm) [inline tests/async_engine/test_api_server.py:13]: Remember to update the values 0 here and other places after we have switched to use "enable" and "disable".

**Comment by robertgshaw2-neuralmagic** [comment]
Closing in favor of https://github.com/neuralmagic/nm-vllm/pull/297

**Pull request closed without merging**